include/Surelog/Design/ModuleInstance.h

```diff
@@ -56,9 +56,9 @@ class ModuleInstance : public ValuedComponentI {
   void setInstanceBinding(ModuleInstance* boundToInstance) {
     m_boundInstance = boundToInstance;
   }
-  DesignComponent* getDefinition() { return m_definition; }
-  unsigned int getNbChildren() { return m_allSubInstances.size(); }
-  ModuleInstance* getChildren(unsigned int i) {
+  DesignComponent* getDefinition() const { return m_definition; }
```

Comment:
I recommend not using a const method when we return a non-const pointer, because it breaks the idea of `const`.

`const` indicates that the state of the object does not change, but since now this returns a non-const pointer, we can change the state of the object through the backdoor -- now even if we pass a const reference of `ModuleInstance`, we don't know if it is not modified because it is possible to retrieve a non-const pointer.

So only make a method const when it does not change the state of the object either directly or indirectly by returning a non-const pointer.

(if we have both use-cases, then I'd usually have two methods: 

```c++
const DesignComponent* getDefinition() const { ... }
DesignComponent* getMutableDefinition() { ... }
```

Reply:
The 'const' here is for an accessory on 'ModuleInstance' whose state isn't changing because of this call. The state of 'Definition' however might change. Every _object_ dictates its own limits in polymorphism.

Reply:
Yes, this is true to the letter, but not the spirit. This is why we should have const and non-const versions of the same thing, otherwise we'll never be able to fix the const-issues as returning non-const pointers from const getters will just plaster over the issue.

I suggest to not include these kind of const changes in this PR, as they  are unrelated.

Reply:
There are lot of const issues (and ownership problems) across the codebase.

Will revert the const changes where applicable.